ui/src/components/table/table-pagination.js

```diff
@@ -64,6 +64,10 @@ export function useTablePaginationState (vm, getCellValue) {
     nextTick(() => {
       emit('request', {
         pagination: prop.pagination || computedPagination.value,
+        // FIXME: 'props.filter' is string/object, but 'prop.filter' can be controlled by the user, and the docs are suggesting 'prop.filter' is a function
+        // So, value of 'filter' becomes function/string/object, which makes a lot of things unpredictable and can break things
+        // Either update the docs to say 'prop.filter' should be a string/object, or use 'prop.filter || props.filterMethod' or maybe get 'computedFilterFunction' here and use that instead of 'props.filterMethod'
+        // The examples on our docs are using 'filter' as a string in onRequest handler, but the JSON API is saying 'filter' is a function
```

Comment:
@rstoenescu can you please confirm if what I have found is correct?

Reply:
The "filter" prop's correct type is String or Object. Never a Function.
The docs instances where "filter" is used as a Function are wrong.

Reply:
So, both `QTable.json > events > request > params > requestProp > definition > filter` and `QTable.json > methods > requestServerInteraction > params > props > definition > filter` should be the same as `QTable.json > props > filter`, right?

(They should have the type of `[ 'String, 'Object' ]` instead of `Function`)

Reply:
Seems correct how it is now. The "request" and "requestServerInteraction" events' "filter" does not refers to the same thing as the "filter" prop. They are convenient methods for applying the filtering.